# StaPH-B Public Health Viral Genomics

## Workflows designed for SARS-CoV-2 genomic analysis
This repository is a collection of workflows hosted by the StaPH-B community to provide a centralized location for SARS-CoV-2 workflow development.
### Workflows
* **
#### Released

#### In Development
* **Titan Illumina PE**: Workflow for processing raw Illumina Fastq Paired End reads and generating consensus genome assemblies with a variety of QC metrics.  

* **Titan Illumina SE**: Workflow for processing raw Illumina Fastq Single End reads and generating consensus genome assemblies with a variety of QC metrics.  

* **Titan ONT**: Workflow for processing raw ONT reads (Fastq or Fast5 format) and generating consensus genome assemblies with a variety of QC metrics.  

* **Titan Clear Labs**: Workflow for processing reads obtained from the Clear Labs sequencing approach and generating consensus genome assemblies with a variety of QC metrics.
* **Mercury**: Workflow for uploading genomic sequence data into [GISAID](https://www.gisaid.org/), [NCBI-GenBank](https://www.ncbi.nlm.nih.gov/genbank/), and [NCBI-SRA](https://www.ncbi.nlm.nih.gov/sra/)
* **
### Contributors
* Workflows originally derived from [Theiagen](http://www.theiagen.com/)'s:
  * [Public Health Viral Genomics](https://github.com/theiagen/public_health_viral_genomics) repository.
* Design influenced by UPHL's [Cecret](https://github.com/UPHL-BioNGS/Cecret)
* Design influenced by StaPH-B's [Monroe](https://staph-b.github.io/staphb_toolkit/workflow_docs/monroe/)
